# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-am-build gpgme-build libmetalink-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists libpsl-build && REQUIRES="$REQUIRES libpsl-build"
pkg-config --exists pcre-build && REQUIRES="$REQUIRES pcre-build"

if pkg-config --exists openssl-build; then
  SSL=--with-ssl=openssl
  REQUIRES="$REQUIRES openssl-build"
else
  SSL=--with-ssl=no
fi

# configure (ac-2.70; am-1.16)
aclocal -I m4; autoheader; autoconf; automake -a -c
./configure $SSL --prefix=

# build
make

# install
install -d $DESTDIR/bin
install src/wget $DESTDIR/bin

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/wget-build.pc << EOF
Name: wget
Version: 1.21.1
Description: GNU Wget
Requires: $REQUIRES

exec=\\
/bin/wget
EOF

exit

# patch
--- wget-1.21.1.orig/configure.ac
+++ wget-1.21.1/configure.ac
@@ -48,6 +48,8 @@
 dnl
 AM_INIT_AUTOMAKE([subdir-objects dist-lzip])
 
+AM_MAINTAINER_MODE
+
 AC_USE_SYSTEM_EXTENSIONS
 AC_PROG_CC
 AM_PROG_CC_C_O
@@ -338,8 +340,6 @@
 AM_ICONV
 gl_INIT
 
-AX_CODE_COVERAGE
-
 dnl
 dnl Checks for library functions.
 dnl
--- wget-1.21.1.orig/doc/texi2pod.pl
+++ wget-1.21.1/doc/texi2pod.pl
@@ -1,4 +1,4 @@
-#! /usr/bin/env perl
+#! /bin/env perl
 
 #   Copyright (C) 1999-2001, 2003, 2007, 2009-2011, 2015, 2018-2021 Free
 #   Software Foundation, Inc.
--- wget-1.21.1.orig/lib/tempname.c
+++ wget-1.21.1/lib/tempname.c
@@ -287,7 +287,8 @@
 
   for (count = 0; count < attempts; ++count)
     {
-      for (size_t i = 0; i < x_suffix_len; i++)
+      size_t i;
+      for (i = 0; i < x_suffix_len; i++)
         {
           if (vdigits == 0)
             {
--- wget-1.21.1.orig/src/progress.c
+++ wget-1.21.1/src/progress.c
@@ -1375,7 +1375,8 @@
 #endif
   if (params)
     {
-      for (const char *param = params; *param; )
+      const char *param;
+      for (param = params; *param; )
         {
           if (!strncmp (param, "force", 5))
             current_impl_locked = 1;

# source
https://ftp.gnu.org/gnu/wget/wget-1.21.1.tar.gz
